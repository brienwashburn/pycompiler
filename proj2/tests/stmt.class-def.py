class _1: pass
class _2(): pass
class _3(First): pass
class _4(First,): pass
class _5(First, Second, Third): pass
class _6(First, Second, Third,): pass
class _7(First, *Second): pass
class _8(First, Second=foo, Third=bar): pass
class _9(First=foo, *Second, Third=quux, **Fourth): pass
class _10(**First): pass
class _11(*First, second=foo, Third=bar, **Fifth): pass
class _12(*First): pass
class _13(lowercase): pass
class _14(PreFirst, First=foo, *Second, Third=quux, **Fourth): pass

class _15(PreFirst, First=foo, *Second, Third=quux, **Fourth):
    a = 1
    b = 2
    c = 3
    a * b * c

@_16d
class _16: pass

@_17d.a.b.c
class _17: pass

@_18d(First)
class _18: pass

@_19d(First,)
class _19: pass

@_20d(First, Second, Third)
class _20: pass

@_21d(PreFirst, First=foo, *second, third=quux, **fourth)
class _21: pass

@_22d.a.b.c(PreFirst, First=foo, *second, third=quux, **fourth)
class _22: pass

@_23d1
@_23d2.a.b.c(PreFirst, First=foo, *second, third=quux, **fourth)
@_23_rax(first)
class _23: pass

@_24d1
@_24d2.a.b.c(PreFirst, First=foo, *second, third=quux, **fourth)
class _24(PreFirst, First=foo, *Second, Third=quux, **Fourth):
    a = 1
    b = 2
    c = 3
    a * b * c
