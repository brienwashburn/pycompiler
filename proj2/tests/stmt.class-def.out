(Module
 (ClassDef
  (name _1)
  (bases)
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _2)
  (bases)
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _3)
  (bases (Name First))
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _4)
  (bases (Name First))
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _5)
  (bases (Name First) (Name Second) (Name Third))
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _6)
  (bases (Name First) (Name Second) (Name Third))
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _7)
  (bases (Name First))
  (keywords)
  (starargs (Name Second))
  (kwargs #f)
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _8)
  (bases (Name First))
  (keywords (Second (Name foo)) (Third (Name bar)))
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _9)
  (bases)
  (keywords (First (Name foo)) (Third (Name quux)))
  (starargs (Name Second))
  (kwargs (Name Fourth))
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _10)
  (bases)
  (keywords)
  (starargs #f)
  (kwargs (Name First))
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _11)
  (bases)
  (keywords (second (Name foo)) (Third (Name bar)))
  (starargs (Name First))
  (kwargs (Name Fifth))
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _12)
  (bases)
  (keywords)
  (starargs (Name First))
  (kwargs #f)
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _13)
  (bases (Name lowercase))
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _14)
  (bases (Name PreFirst))
  (keywords (First (Name foo)) (Third (Name quux)))
  (starargs (Name Second))
  (kwargs (Name Fourth))
  (body (Pass))
  (decorator_list))
 (ClassDef
  (name _15)
  (bases (Name PreFirst))
  (keywords (First (Name foo)) (Third (Name quux)))
  (starargs (Name Second))
  (kwargs (Name Fourth))
  (body
   (Assign (targets (Name a)) (value (Num 1)))
   (Assign (targets (Name b)) (value (Num 2)))
   (Assign (targets (Name c)) (value (Num 3)))
   (Expr (BinOp (BinOp (Name a) Mult (Name b)) Mult (Name c))))
  (decorator_list))
 (ClassDef
  (name _16)
  (bases)
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list (Name _16d)))
 (ClassDef
  (name _17)
  (bases)
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list (Attribute (Attribute (Attribute (Name _17d) a) b) c)))
 (ClassDef
  (name _18)
  (bases)
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list
   (Call
    (func (Name _18d))
    (args (Name First))
    (keywords)
    (starargs #f)
    (kwargs #f))))
 (ClassDef
  (name _19)
  (bases)
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list
   (Call
    (func (Name _19d))
    (args (Name First))
    (keywords)
    (starargs #f)
    (kwargs #f))))
 (ClassDef
  (name _20)
  (bases)
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list
   (Call
    (func (Name _20d))
    (args (Name First) (Name Second) (Name Third))
    (keywords)
    (starargs #f)
    (kwargs #f))))
 (ClassDef
  (name _21)
  (bases)
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list
   (Call
    (func (Name _21d))
    (args (Name PreFirst))
    (keywords (First (Name foo)) (third (Name quux)))
    (starargs (Name second))
    (kwargs (Name fourth)))))
 (ClassDef
  (name _22)
  (bases)
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list
   (Call
    (func (Attribute (Attribute (Attribute (Name _22d) a) b) c))
    (args (Name PreFirst))
    (keywords (First (Name foo)) (third (Name quux)))
    (starargs (Name second))
    (kwargs (Name fourth)))))
 (ClassDef
  (name _23)
  (bases)
  (keywords)
  (starargs #f)
  (kwargs #f)
  (body (Pass))
  (decorator_list
   (Name _23d1)
   (Call
    (func (Attribute (Attribute (Attribute (Name _23d2) a) b) c))
    (args (Name PreFirst))
    (keywords (First (Name foo)) (third (Name quux)))
    (starargs (Name second))
    (kwargs (Name fourth)))
   (Call
    (func (Name _23_rax))
    (args (Name first))
    (keywords)
    (starargs #f)
    (kwargs #f))))
 (ClassDef
  (name _24)
  (bases (Name PreFirst))
  (keywords (First (Name foo)) (Third (Name quux)))
  (starargs (Name Second))
  (kwargs (Name Fourth))
  (body
   (Assign (targets (Name a)) (value (Num 1)))
   (Assign (targets (Name b)) (value (Num 2)))
   (Assign (targets (Name c)) (value (Num 3)))
   (Expr (BinOp (BinOp (Name a) Mult (Name b)) Mult (Name c))))
  (decorator_list
   (Name _24d1)
   (Call
    (func (Attribute (Attribute (Attribute (Name _24d2) a) b) c))
    (args (Name PreFirst))
    (keywords (First (Name foo)) (third (Name quux)))
    (starargs (Name second))
    (kwargs (Name fourth))))))
